src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatGeneratorStatics.cs

```diff
@@ -249,6 +249,20 @@ internal static MethodInfo StoreCollectionMemberInfoMethod
             }
         }
 
+        private static MethodInfo s_resetCollectionMemberInfoMethod;
+        internal static MethodInfo ResetCollectionMemberInfoMethod
+        {
+            get
+            {
+                if (s_resetCollectionMemberInfoMethod == null)
+                {
+                    s_resetCollectionMemberInfoMethod = typeof(XmlObjectSerializerReadContext).GetMethod("ResetCollectionMemberInfo", Globals.ScanAllMembers, new Type[] { });
+                    Debug.Assert(s_resetCollectionMemberInfoMethod != null);
```

Comment:
Does the nameof() operator do the right thing if you specify a method in another class? If it does, I would prefer that to an inline string constant.